System/Random.hs

```diff
@@ -1002,7 +1003,13 @@ instance Random Double where
   random = randomDouble
   randomM = uniformR (0, 1)
 
-instance UniformRange Double
+instance UniformRange Double where
+#if __GLASGOW_HASKELL__ >= 844
```

Comment:
If we're not providing an alternative implementation in an `#else` branch, I would wrap the entire instance declaration in the `#if` (i.e., move it up one line). Otherwise we get compilation warnings about missing methods when the condition is not fulfilled.